basiclibrary/src/main/java/basiclibrary/Lab03.java

```diff
@@ -1,13 +1,51 @@
 package basiclibrary;
 
-import java.util.ArrayList;
+import java.util.*;
 
 public class Lab03 {
   public static String weatherData(int[][] weatherArr) {
-    return "To be completed";
+    HashSet<Integer> weatherSet = new HashSet<>();
+
+    for (int[] ints : weatherArr) {
+      for (int day : ints) {
+        weatherSet.add(day);
+      }
+    }
+
+    if (weatherSet.isEmpty()) return "0";
+
+    int min = Collections.min(weatherSet);
+    int max = Collections.max(weatherSet);
+    StringBuilder response = new StringBuilder();
+    response.append("High: ").append(max).append("\n");
+    response.append("Low: ").append(min).append("\n");
+
+    for (int i = min; i < max; i++) {
+      if (!weatherSet.contains(i)) response.append("Never saw temperature: ").append(i).append("\n");
+    }
+
+    return response.toString();
   }
 
-  public static String tally(ArrayList<String> votes) {
-    return "To be completed";
+  public static String tally(List<String> votes) {
+    if (votes.isEmpty()) return "0";
+
+    HashSet<String> votedItems = new HashSet<>();
+    votedItems.addAll(votes);
+
+    int numVotes = 0;
+    String winner = null;
+
+    for (String item : votedItems) {
+      int count = Collections.frequency(votes, item);
```

Comment:
This is certainly convenient, but `Collections.frequency` is O(n) time - many of the built in methods are. This makes your `tally` method have an O(n^2) time. Moving forward I encourage you to spend some time thinking of different ways to approach a problem, and how efficient they are.